panic: interface conversion: ast.Expr is *ast.SelectorExpr, not *ast.Ident

The linter panics on the following code
```go
func Example(service protoreflect.ServiceDescriptor) {
	for i := 0; i < service.Methods().Len(); i++ {
		print(i)
	}
}
```

```
panic: interface conversion: ast.Expr is *ast.SelectorExpr, not *ast.Ident

goroutine 70 [running]:
github.com/ckaznocha/intrange.findNExpr({0x138cfc8?, 0xc000216340?})
        external/gazelle~~go_deps~com_github_ckaznocha_intrange/intrange.go:237 +0x13a
github.com/ckaznocha/intrange.run.check.func1({0x138c4e0?, 0xc0002163c0})
        external/gazelle~~go_deps~com_github_ckaznocha_intrange/intrange.go:95 +0x1f4
golang.org/x/tools/go/ast/inspector.(*Inspector).Preorder(0xc000010228, {0xc00022ed18?, 0x1538860?, 0xc000052908?}, 0xc00022ed08)
        external/gazelle~~go_deps~org_golang_x_tools/go/ast/inspector/inspector.go:82 +0x8f
github.com/ckaznocha/intrange.run(0xc0001a8270)
        external/gazelle~~go_deps~com_github_ckaznocha_intrange/intrange.go:47 +0x129
main.(*action).execOnce(0xc0001ade00)
        external/rules_go~/go/tools/builders/nogo_main.go:354 +0x8b7
sync.(*Once).doSlow(0x0?, 0xc000123200?)
        GOROOT/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        GOROOT/src/sync/once.go:65
main.(*action).exec(...)
        external/rules_go~/go/tools/builders/nogo_main.go:281
main.execAll.func1(0x0?)
        external/rules_go~/go/tools/builders/nogo_main.go:275 +0x72
created by main.execAll in goroutine 1
        external/rules_go~/go/tools/builders/nogo_main.go:273 +0x48
```

--- a/intrange.go
+++ b/intrange.go
@@ -234,15 +234,11 @@ func check(pass *analysis.Pass) func(node ast.Node) {
 func findNExpr(expr ast.Expr) ast.Expr {
 	switch e := expr.(type) {
 	case *ast.CallExpr:
-		if e.Fun.(*ast.Ident).Name != "len" {
-			return nil
+		if fun, ok := e.Fun.(*ast.Ident); ok && fun.Name == "len" && len(e.Args) == 1 {
+			return findNExpr(e.Args[0])
 		}
 
-		if len(e.Args) != 1 {
-			return nil
-		}
-
-		return findNExpr(e.Args[0])
+		return nil
 	case *ast.BasicLit:
 		return nil
 	case *ast.Ident:

--- a/testdata/go.mod
+++ b/testdata/go.mod
@@ -1,3 +1,5 @@
 module github.com/ckaznocha/intrange/testdata
 
 go 1.21
+
+require google.golang.org/protobuf v1.33.0

--- a/testdata/main.go
+++ b/testdata/main.go
@@ -1,6 +1,10 @@
 package main
 
-import "testing"
+import (
+	"testing"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
 
 func main() {
 	for i := 2; i < 10; i++ {
@@ -185,5 +189,12 @@ func main() {
 
 	for i := 0; i < len(t2.m); i++ {
 		t2.m[4] = 4
+	}
+}
+
+// https://github.com/ckaznocha/intrange/issues/16
+func issue16(service protoreflect.ServiceDescriptor) {
+	for i := 0; i < service.Methods().Len(); i++ { // want `for loop can be changed to use an integer range \(Go 1\.22\+\)`
+		print(i)
 	}
 }

